[tests] added forms test

radekdoulik opened the pull request:
- to measure UI app startup times

 - added RunUITests task (similar to unInstrumentationTests) to start
   app/activity

  - renamed .*UnitTestApk.* to .*TestApk.* as it handles also the new
   UI test

  - updated ProcessLogcatTiming to be able to process logcat output of
    started Android activity process

radekdoulik commented:
the build will fail as it will need a fix from https://github.com/xamarin/xamarin-android/pull/815 to be able to use custom timing definitions

jonpryor commented:
PR #515 has been merged.

Indentation is...weirdly all over the map. It should be corrected.

jonpryor commented on src/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/RunUITests.cs:
```diff
@@ -0,0 +1,37 @@
+		public                  string              AdbTarget                   { get; set; }
+		public                  string              AdbOptions                  { get; set; }
+
+        [Required]
```
This should be left-aligned with the `public` on the following line.

jonpryor commented on tests/forms/App.xaml.cs:
```diff
@@ -0,0 +1,27 @@
+
+namespace forms
+{
+    public partial class App : Application
```
I take it that the default Xamarin.Forms template uses spaces? Not sure if this needs to be fixed, just seems weird.

jonpryor commented on tests/forms/Droid/MainActivity.cs:
```diff
@@ -0,0 +1,65 @@
+    public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
+    {
+        bool firstOnCreate = true;
+		bool firstOnStart = true;
```
The `bool` on this and the following line should be indented the same amount as the previous line.

jonpryor commented on tests/forms/Droid/MainActivity.cs:
```diff
@@ -0,0 +1,65 @@
+                Console.WriteLine("startup-timing: OnCreate end reached");
+                firstOnCreate = false;
+            }
+		}
```
This line is indented too much; it should be aligned with the `{` for `OnCreate()`.

jonpryor commented on tests/forms/Droid/MainActivity.cs:
```diff
@@ -0,0 +1,65 @@
+                Console.WriteLine("startup-timing: OnStart end reached");
+                firstOnStart = false;
+            }
+		}
```
This line is indented too much; it should be aligned with the `{` for `OnStart()`.

jonpryor commented on tests/forms/Droid/MainActivity.cs:
```diff
@@ -0,0 +1,65 @@
+        {
+            if (firstOnStart)
+                Console.WriteLine("startup-timing: OnStart reached");
+			base.OnStart();
```
This line is indented too much; it should be aligned with the `if`, two lines above.

jonpryor commented on tests/forms/Droid/Resources/values/styles.xml:
```diff
@@ -0,0 +1,27 @@
+		<!-- colorPrimaryDark is used for the status bar -->
+		<item name="colorPrimaryDark">#1976D2</item>
+		<!-- colorAccent is used as the default value for colorControlActivated
+         which is used to tint widgets -->
```
What made *this* line wrap decision? It's ugly. :-/

jonpryor commented on tests/forms/Services/CloudDataStore.cs:
```diff
@@ -0,0 +1,83 @@
+        public async Task<Item> GetItemAsync(string id)
+        {
+            if (id != null && CrossConnectivity.Current.IsConnected)
+            {
```
Ditto.

jonpryor commented on tests/forms/ViewModels/AboutViewModel.cs:
```diff
@@ -0,0 +1,19 @@
+        {
+            Title = "About";
+
+            OpenWebCommand = new Command(() => Device.OpenUri(new Uri("https://xamarin.com/platform")));
```
Space ` ` before `(`.

jonpryor commented on tests/forms/ViewModels/BaseViewModel.cs:
```diff
@@ -0,0 +1,53 @@
+
+        bool isBusy = false;
+        public bool IsBusy
+        {
```
This should be on the previous line.

jonpryor commented on tests/forms/ViewModels/BaseViewModel.cs:
```diff
@@ -0,0 +1,53 @@
+
+        string title = string.Empty;
+        public string Title
+        {
```
Ditto.

jonpryor commented on tests/forms/ViewModels/ItemsViewModel.cs:
```diff
@@ -0,0 +1,55 @@
+            LoadItemsCommand = new Command(async () => await ExecuteLoadItemsCommand());
+
+            MessagingCenter.Subscribe<NewItemPage, Item>(this, "AddItem", async (obj, item) =>
+            {
```
This `{` should be on the previous line.

jonpryor commented on tests/forms/ViewModels/ItemsViewModel.cs:
```diff
@@ -0,0 +1,55 @@
+        {
+            Title = "Browse";
+            Items = new ObservableCollection<Item>();
+            LoadItemsCommand = new Command(async () => await ExecuteLoadItemsCommand());
```
Space ` ` before `(`.

jonpryor commented on tests/forms/ViewModels/ItemsViewModel.cs:
```diff
@@ -0,0 +1,55 @@
+            });
+        }
+
+        async Task ExecuteLoadItemsCommand()
```
Space ` ` before `(`.

jonpryor commented on tests/forms/Views/MainPage.cs:
```diff
@@ -0,0 +1,53 @@
+            {
+                case Device.iOS:
+                    itemsPage = new NavigationPage(new ItemsPage())
+                    {
```
This `{` should be on the previous line.

jonpryor commented:
Overall, I don't like the name `tests/forms`. It doesn't have any semantic meaning, to answer "why does this directory exist?" (Compare to `tests/CodeGen-*`, which deals with code generation, or `tests/ResolveImports`, which deals with project resolution.) "forms" is a *what*, not a *why*.

I would prefer `tests/Xamarin.Forms-Performace-Integration`, or something otherwise semantically meaningful.

radekdoulik commented:
rebased on current master, so hopefully it will build OK now.

regenerated the test with mono coding style set in VisualStudio/Mac, so hopefully it looks better now. also renamed the test directory and package name as suggested.

jonpryor commented:
build

jonpryor commented:
One downside to renaming `@(UnitTestApk)` to `@(TestApk)` is that this will require additional monodroid-side changes in order to merge. Is this desirable?

jonpryor commented on tests/Xamarin.Forms-Performance-Integration/Xamarin.Forms.Performance.Integration.sln:
```diff
@@ -0,0 +1,19 @@
+
```
This file shouldn't exist. Instead, the .csproj should be added to the top level Xamarin.Android-Tests.sln file.

radekdoulik replied:
Good catch, updated.

radekdoulik commented:
The monodroid changes are simple, created https://github.com/xamarin/monodroid/pull/665 to do that. It can be completed with adding XA bump.
